packages/lesswrong/server/analytics/analyticsCron.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console */
+import moment from "moment";
+import { addCronJob } from "../cronUtil";
+import { Globals } from "../vulcan-lib";
+import PostViewsRepo from "../repos/PostViewsRepo";
+import PostViewTimesRepo from "../repos/PostViewTimesRepo";
+import IncrementalViewRepo from "../repos/IncrementalViewRepo";
+import { isEAForum } from "../../lib/instanceSettings";
+
+async function updateDailyAnalyticsCollection<N extends CollectionNameString>({
+  repo,
+  earliestStartDate,
+  latestEndDate,
+}: {
+  repo: IncrementalViewRepo<N>;
+  earliestStartDate: Date;
+  latestEndDate: Date;
+}) {
+  const { earliestWindowStart, latestWindowEnd } = await repo.getDateBounds()
+
+  // `ranges` will be a list of days to recalculate the data for. When this is running as a cron job in steady state,
+  // this will just be the most recent day. In general it will be all days between `earliestStartDate` and `latestEndDate`
+  // that are not already covered by the range [`earliestWindowStart`, `latestWindowEnd`)
+  const ranges: {startDate: Date, endDate: Date}[] = []
+
+  let currentDate = moment(earliestStartDate).startOf('day').toDate();
+
+  // Ensure we cover the endDate if it's the same as latestWindowEnd
+  const inclusiveEndDate = latestWindowEnd && moment(latestWindowEnd).subtract(1, 'day').toDate();
+
+  while (currentDate <= latestEndDate) {
+    // Skip dates that are within the [`earliestWindowStart`, `latestWindowEnd`) interval
```

Comment:
As above:
> In this and below, there is a subtle point that we do always recalculate the most recent day, because it may have had partial data. Agree this comment is useful, I'll just reword it to make sure this is clear